api/app/src/providers/fitbit.ts

```diff
@@ -332,4 +335,71 @@ export class Fitbit extends Provider implements OAuth2 {
       }
     );
   }
+
+  async postAuth(token: string, userId?: string) {
+    console.log("Post auth started");
+    const accessToken = JSON.parse(token).access_token;
+
+    const scopes = JSON.parse(token).scope;
+
+    const subscriptionId = userId;
+
+    if (this.allRequiredScopesIncluded(scopes)) {
+      const subscriptionUrl = `${Fitbit.URL}/${Fitbit.API_PATH}/apiSubscriptions/${subscriptionId}.json`;
+      this.createSubscription(subscriptionUrl, accessToken);
+      console.log("All scopes included. User subscribed to all Fitbit WH collectionTypes.");
+    } else {
+      if (scopes.includes("activity")) {
+        const subscriptionUrl = `${Fitbit.URL}/${Fitbit.API_PATH}/activities/apiSubscriptions/${subscriptionId}activities.json`;
+        await this.createSubscription(subscriptionUrl, accessToken);
+      }
+      if (scopes.includes("nutrition")) {
+        const subscriptionUrl = `${Fitbit.URL}/${Fitbit.API_PATH}/foods/apiSubscriptions/${subscriptionId}foods.json`;
+        await this.createSubscription(subscriptionUrl, accessToken);
+      }
+      if (scopes.includes("weight")) {
+        const subscriptionUrl = `${Fitbit.URL}/${Fitbit.API_PATH}/body/apiSubscriptions/${subscriptionId}body.json`;
+        await this.createSubscription(subscriptionUrl, accessToken);
+      }
+      if (scopes.includes("sleep")) {
+        const subscriptionUrl = `${Fitbit.URL}/${Fitbit.API_PATH}/sleep/apiSubscriptions/${subscriptionId}sleep.json`;
+        await this.createSubscription(subscriptionUrl, accessToken);
+      }
+    }
+
+    // TODO: userRevokedAccess. Revoking the token makes it so the webhook do not trigger anything on our end. So, perhaps, this isn't necessary at all.
+    // const userRevokedAccessUrl = `${Fitbit.URL}/${Fitbit.API_PATH}/userRevokedAccess/apiSubscriptions/${subscriptionId}sleep.json`;
+    // await this.createSubscription(userRevokedAccessUrl, accessToken);
+  }
+
+  // Checks if all of the required scopes were authorized by the user
+  allRequiredScopesIncluded(scopes: string): boolean {
+    const allRequiredScopes = "activity, nutrition, profile, settings, sleep, weight";
+
+    for (const scope of allRequiredScopes.split(",")) {
+      if (!scopes.includes(scope)) return false;
+    }
+    return true;
+  }
+
+  // Creates a subscription for all or one specific collectionType
+  async createSubscription(url: string, accessToken: string): Promise<void> {
+    try {
+      const resp = await axios.post(url, null, {
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${accessToken}`,
+          "Content-Length": 0,
+        },
+      });
+      console.log("Fitbit WH subscription created successfully.", resp.data);
+    } catch (error) {
+      console.log("Fitbit post auth failed.");
+      capture.error(error, {
+        extra: { context: `fitbit.postAuth`, url },
+      });
+
+      throw new Error(`WH subscription failed Fitbit. Cause: ${error}`);
```

Comment:
It would be better to set the cause as object than passing it on the error message, something like this, IIRC:

```ts
throw new Error(`WH subscription failed Fitbit`, {cause: error});
```